system_tests/speech.py

```diff
@@ -0,0 +1,117 @@
+    This is a mutable stand-in to allow test set-up to modify
+    global state.
+    """
+    CLIENT = None
+    TEST_BUCKET = None
+
+
+def setUpModule():
+    Config.CLIENT = speech.Client()
+    # Now create a bucket for GCS stored content.
+    storage_client = storage.Client()
+    bucket_name = 'new' + unique_resource_id()
+    Config.TEST_BUCKET = storage_client.bucket(bucket_name)
+    # 429 Too Many Requests in case API requests rate-limited.
+    retry_429 = RetryErrors(exceptions.TooManyRequests)
+    retry_429(Config.TEST_BUCKET.create)()
+
+
+def tearDownModule():
+    # 409 Conflict if the bucket is full.
+    # 429 Too Many Requests in case API requests rate-limited.
+    bucket_retry = RetryErrors(
+        (exceptions.TooManyRequests, exceptions.Conflict))
+    bucket_retry(Config.TEST_BUCKET.delete)(force=True)
+
+
+class TestSpeechClient(unittest.TestCase):
+    def setUp(self):
+        self.to_delete_by_case = []
+
+    def tearDown(self):
+        for value in self.to_delete_by_case:
+            value.delete()
+
+    def _make_sync_request(self, content=None, source_uri=None,
+                           max_alternatives=None):
+        client = Config.CLIENT
+        sample = client.sample(content=content,
+                               source_uri=source_uri,
+                               encoding=speech.Encoding.LINEAR16,
+                               sample_rate=16000)
+        result = client.sync_recognize(sample,
+                                       language_code='en-US',
+                                       max_alternatives=max_alternatives,
+                                       profanity_filter=True,
+                                       speech_context=['Google',
+                                                       'cloud'])
+        return result
+
+    def _check_best_results(self, results):
+        from google.cloud.speech.transcript import Transcript
+
+        top_result = results[0]
+        self.assertIsInstance(top_result, Transcript)
+        self.assertEqual(top_result.transcript,
+                         'hello thank you for using Google Cloud platform')
+        self.assertGreater(top_result.confidence, 0.90)
+
+    def test_sync_recognize_local_file(self):
+        import os
+        from google.cloud.speech.transcript import Transcript
+
+        file_name = os.path.join('system_tests', 'data', 'hello.wav')
+
+        with open(file_name, 'rb') as file_obj:
+            results = self._make_sync_request(content=file_obj.read(),
+                                              max_alternatives=2)
+            second_alternative = results[1]
+            self.assertEqual(len(results), 2)
+            self._check_best_results(results)
+            self.assertIsInstance(second_alternative, Transcript)
+            self.assertEqual(second_alternative.transcript,
+                             'thank you for using Google Cloud platform')
```

Comment:
Maybe `'thank you for using Google Cloud platform'` should be a class constant since you use it here and elsewhere. The you could say

```python
self.assertEqual(second_alternative.transcript, self.FOO)

# AND above

self.assertEqual(top_result.transcript, 'hello ' + self.FOO)
```